web-common/src/features/canvas/CanvasDashboardEmbed.svelte

```diff
@@ -35,33 +35,43 @@
   $: canvasData = $canvasResolverQuery.data;
 </script>
 
-<CanvasDashboardWrapper
-  {maxWidth}
-  filtersEnabled={canvas?.spec?.filtersEnabled}
->
-  {#each rows as { items = [], height = MIN_HEIGHT, heightUnit = "px" }, rowIndex (rowIndex)}
-    {@const widths = normalizeSizeArray(items?.map((el) => el?.width ?? 0))}
-    {@const types = items?.map(
-      ({ component }) =>
-        canvasData?.resolvedComponents?.[component ?? ""]?.component?.spec
-          ?.renderer,
-    )}
-    <RowWrapper
-      {maxWidth}
-      {rowIndex}
-      zIndex={50 - rowIndex * 2}
-      height="{height}{heightUnit}"
-      gridTemplate={widths.map((w) => `${w}fr`).join(" ")}
-    >
-      {#each items as item, columnIndex (columnIndex)}
-        <ItemWrapper type={types[columnIndex]} zIndex={4 - columnIndex}>
-          <CanvasComponent canvasItem={item} id={item.component ?? ""} />
-        </ItemWrapper>
-      {/each}
-    </RowWrapper>
-  {:else}
-    <div class="size-full flex items-center justify-center">
-      <p class="text-lg text-gray-500">No components added</p>
-    </div>
-  {/each}
-</CanvasDashboardWrapper>
+{#if canvasName}
+  <CanvasDashboardWrapper
+    {maxWidth}
+    filtersEnabled={canvas?.spec?.filtersEnabled}
+    {canvasName}
+  >
+    {#each rows as { items = [], height = MIN_HEIGHT, heightUnit = "px" }, rowIndex (rowIndex)}
+      {@const widths = normalizeSizeArray(items?.map((el) => el?.width ?? 0))}
+      {@const types = items?.map(
+        ({ component }) =>
+          canvasData?.resolvedComponents?.[component ?? ""]?.component?.spec
+            ?.renderer,
```

Comment:
For this component, we can simplify some of these references by using the `useCanvas` selector available in `features/canvas/selector.ts`. This also makes sure that we don't use different instances of the same query.